src/messageComposer/messageComposer.ts

```diff
@@ -337,6 +341,26 @@ export class MessageComposer {
     composition?: DraftResponse | MessageResponse | LocalMessage,
   ) => initEditingAuditState(composition);
 
+  takeStateSnapshot = (): MessageComposerStateSnapshot => {
+    const textComposerState = this.textComposer.state.getLatestValue();
+    const attachmentManagerState = this.attachmentManager.state.getLatestValue();
+    const linkPreviewsManagerState = this.linkPreviewsManager.state.getLatestValue();
+    const pollComposerState = this.pollComposer.state.getLatestValue();
+    const customDataManagerState = this.customDataManager.state.getLatestValue();
+    const state = this.state.getLatestValue();
+
+    return {
+      restore: () => {
+        this.state.next(state);
+        this.textComposer.state.next(textComposerState);
+        this.attachmentManager.state.next(attachmentManagerState);
+        this.linkPreviewsManager.state.next(linkPreviewsManagerState);
+        this.pollComposer.state.next(pollComposerState);
+        this.customDataManager.state.next(customDataManagerState);
+      },
```

Comment:
What would happen if something modifies the state, but a transaction fails and it is rolled back?
Would we lose data in that case?

Reply:
What is you are concerned about?

Reply:
If you are concerned about state changes due to some effects we could add something line `messageComposer.lockState()` and that could be called before `messageComposer. takeStateSnapshot()`.

Reply:
Do we need this functionality in the state store? I'm not sure if introducing global locks is necessary.
Can we do this in the UI SDK only?